cfme/utils/appliance/implementations/ui.py

```diff
@@ -133,11 +161,15 @@ def make_document_focused(self):
 
     def ensure_page_safe(self, timeout='20s'):
         # THIS ONE SHOULD ALWAYS USE JAVASCRIPT ONLY, NO OTHER SELENIUM INTERACTION
+        if self.browser.page_dirty:
+                if (self.browser.alert_present and
```

Comment:
This indentation looks too far?

Reply:
Consider implementing this as single set of statements:

```
if (self.browser.page_dirty and
       self.browser.alert_present
       self.browser.get_alert().text == 'Abandon changes?'):
    self.browser.handle_alert()
```

Reply:
it is already implemented